server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java

```diff
@@ -324,6 +324,43 @@ public void testEmptyDatasource()
     );
   }
 
+  @Test
+  public void testNewlineDatasource()
+  {
+    Map<String, Object> parser = jsonMapper.convertValue(
+        new StringInputRowParser(
+            new JSONParseSpec(
+                new TimestampSpec("time", "auto", null),
+                new DimensionsSpec(
+                    DimensionsSpec.getDefaultSchemas(ImmutableList.of("time", "dimA", "dimB", "col2")),
+                    ImmutableList.of("dimC"),
+                    null
+                ),
+                null,
+                null
+            ),
+            null
+        ), JacksonUtils.TYPE_REFERENCE_MAP_STRING_OBJECT
+    );
+
+    expectedException.expect(CoreMatchers.instanceOf(IllegalArgumentException.class));
+    expectedException.expectMessage(
+        "dataSource cannot contain the '/n' character."
+    );
+
+    DataSchema schema = new DataSchema(
+        "new\nline\ncharacter",
+        parser,
+        new AggregatorFactory[]{
+            new DoubleSumAggregatorFactory("metric1", "col1"),
+            new DoubleSumAggregatorFactory("metric2", "col2"),
```

Comment:
Could also pass in `null` for `aggregators` to simplify the test code. Code seems to allow it even though the documentation says it's required.